window/src/os/xdg_desktop_portal.rs

```diff
@@ -26,6 +27,18 @@ trait PortalSettings {
     fn SettingChanged(&self, namespace: &str, key: &str, value: OwnedValue) -> Result<()>;
 }
 
+struct State {
+    appearance: Option<Appearance>,
```

Comment:
I'm "worried" about edge cases and also about `get_appearance` having surprising behavior if `subscribe` is never called. Prior to this PR `get_appearance` was easy to reason about, but right now there are some implementation details that aren't called out; in particular, it has no way to invalidate a stale value if `subscribe` isn't called, or if the thread has stopped for whatever reason.

What I'd suggest is:

* Track an `Instant` here to show when STATE.appearance was last assigned
* Track a boolean here to show whether the subscribe thread is running

In `get_appearance`, if subscribe is running then the cached appearance value can be read without fear.

Otherwise, if `Instant::elapsed` is within a reasonable threshold (say, 1 second?), then the cached value can be returned, otherwise, call `read_settings` and update the cached value and the time it was assigned.

In the `subscribe` thread, be sure to update the time each time the cached value is updated as well.

That way we can reasonably cache the appearance in the brief window before the subscribe function is able to be set up and called, and also stand a chance of still behaving reasonably if something weird happens to the subscribe thread for whatever reason.